blueman/plugins/applet/TransferService.py

```diff
@@ -23,6 +25,33 @@
     from html import escape
 
 
+@GObject.type_register
+class CommandRunner(threading.Thread, GObject.Object):
```

Comment:
Ah yes, you are correct on idle_add however... I rather than like to see ``GLib.spawn_async`` used and utilize glibs threading and scheduling capabilities instead of adding subprocess plus threading. Something like below should be just fine for this use case.

```python
from gi.repository import GLib

def on_child_exit(pid, *args, **kwargs):
    print("Closing child with pid %s" % pid)
    GLib.spawn_close_pid(pid)

command = ["/usr/bn/sleep", "10"]

try:
    pid, stdin, stdout, stderr = GLib.spawn_async(
        command, flags=GLib.SPAWN_DO_NOT_REAP_CHILD)
    GLib.child_watch_add(pid, on_child_exit)
except GLib.GError as e:
    print(e.message)
```